pyflakes/checker.py

```diff
@@ -734,8 +734,44 @@ def NAME(self, node):
             # arguments, but these aren't dispatched through here
             raise RuntimeError("Got impossible expression context: %r" % (node.ctx,))
 
+    def CONTINUE(self, node):
+        n = node
+        while hasattr(n, 'parent'):
+            n = n.parent
+            if isinstance(n, (ast.While, ast.For)):
+                # Doesn't apply unless it's in the loop itself
+                in_else = False
+                for else_node in n.orelse:
+                    for child in ast.walk(else_node):
+                        if child == node:
+                            in_else = True
+                            break
+                    if in_else:
+                        break
+                else:
+                    return
+            if isinstance(n, (ast.FunctionDef, ast.ClassDef)):
+                break
+            # Handle Try/TryFinally difference in Python < and >= 3.3
+            if hasattr(n, 'finalbody') and isinstance(node, ast.Continue):
+                in_finally = False
+                for finally_node in n.finalbody:
+                    for child in ast.walk(finally_node):
+                        if child == node:
+                            in_finally = True
+                            break
+                    if in_finally:
+                        self.report(messages.ContinueInFinally, node)
+                        return
+        if isinstance(node, ast.Continue):
+            self.report(messages.ContinueOutsideLoop, node)
+        else:  # ast.Break
+            self.report(messages.BreakOutsideLoop, node)
```

Comment:
Rather than comments, I'd look for loops which could be broken out into specific functions. Then the name of the functions can serve as the documentation.

The high-level algorithm here, as I understand it, is "when a continue or break is encountered, it's an error if that's not in a loop". So I might expect to find a function named `isInLoop()`. Then maybe `CONTINUE()` might be something like:

```python
def CONTINUE(self, node):
    if self.isInLoop(node):
        return
    if isinstance(node, ast.Continue):
        self.report(messages.ContinueOutsideLoop, node)
     else:  # ast.Break
        self.report(messages.BreakOutsideLoop, node)
```

Also, the operation, "traverse the parents until finding a node matching some predicate" has become a pretty common operation, so that can be a function too. Just grepping for `parent` I find a few, the poorly named `getParent`, and also `on_conditional_branch()`.